x/stakeibc/keeper/icqcallbacks_withdrawal_balance.go

```diff
@@ -62,32 +58,15 @@ func WithdrawalBalanceCallback(k Keeper, ctx sdk.Context, args []byte, query icq
 		return errorsmod.Wrapf(types.ErrICAAccountNotFound, "no fee account found for %s", chainId)
 	}
 
-	// Determine the stride commission rate to the relevant portion can be sent to the fee account
-	params := k.GetParams(ctx)
-	strideCommissionInt, err := cast.ToInt64E(params.StrideCommission)
+	// Split the withdrawal amount into a rebate, stride fee, and reinvest portion
+	feeInfo, err := k.CheckForCommunityPoolRebate(ctx, chainId, withdrawalBalanceAmount)
```

Comment:
I don't understand the difference between `WithdrawalBalanceCallback` and `WithdrawalRewardBalanceCallback` - why do we discard `feeInfo.RebateAmount` in `WithdrawalBalanceCallback`?

If it's discarded because this callback is only used for non-DYDX zones, might be safe to verify it's zero rather than ignoring it here.

Reply:
WithdrawalBalance is the ICQ which is part of the normal staking flow.  In this case it is looking for DYDX to turn into stDYDX.  WithdrawalRewardBalance is looking for reward tokens in the withdrawal account which need to be swapped first before being returned to the withdrawal account.  In this case looking for USDC in withdrawal.

WithdrawalBalanceCallback gets the amount of DYDX in withdrawal and does the liquid stake.  WithdrawalRewardBalanceCallback gets the amount of USDC and transfers it to the trade ICA.  Eventually it will come back to the withdrawal account having been swapped as DYDX and then the normal further staking flow can happen from there.  The reward one only applies to DYDX at the moment because it is the only one with trade routes yet but the one without reward is part of the core triggers for liquid stake reinvestment functioning

Reply:
The Rebate amount is used when the USDC comes in originally because the rebate in USDC is sent back at that time.  This function is detecting funds which have already been converted to DYDX -- the withdrawalBalanceAmount is in DYDX -- because this is when stride takes its fee.  RebateAmount is ignored in this function but it is used in the other one which is measuring the USDC which has come in as revenue.

Reply:
That's very helpful, thanks @ethan-stride! Makes sense that these are separate. 

Could I recommend we rename them for clarity?

Here's one option that makes sense to me, but open to others:
*  `WithdrawalBalanceCallback` -> `WithdrawalNativeBalanceCallback`
* `WithdrawalRewardBalanceCallback` -> WithdrawalNonNativeRewardBalanceCallback`

Reply:
We had been using the word "Reward" denom to specifically mean revenue tokens which need to be swapped as compared to the "HostDenom" which is what most chains put into the withdrawal account directly because they inflate the token native to the hostzone.  This terminology is used through the traderoute and hostzone datastructs

So I would suggest `WithdrawalBalanceCallback` -> `WithdrawalHostDenomBalanceCallback` if we want to change it at all but definitely don't include the word reward in that one.  It is part of the route which is most core to all liquid staking stride logic so I am fine with it being the very short name it has now.  And then for the other one, rewards are implicitly in another denom than the host denom so no need to include NonNative

Reply:
I find it confusing because the SDK uses "rewards" as the standard name for all staking rewards, which are typically in the native denom. E.g. claim rewards usually means claim native token rewards.

Reply:
We should probably do a refactor and name what we had called "rewards" to "revenue" instead.  Then USDC would be the revenue denom, and all other chains would get their rewards in the host denom

Reply:
agreed on all the points above. I see how reward denom is a bit confusing but, as Ethan mentioned, it is common terminology across the whole trade route feature so I'd prefer to not do a large refactor across everything right now

My vote would either be to rename `WithdrawalBalanceCallback` to `WithdrawalHostDenomBalanceCallback` as Ethan suggested, or to just leave it as is as the short name `WithdrawalBalanceCallback` since it's a fundamental part of the core liquid staking flow

---
> confusing because the SDK uses "reward" as the standard name for all staking rewards, which are typically in the native denom

Fwiw, it seems like the confusion is more a byproduct of dydx collecting rewards in a USDC, and less a byproduct of our naming. I would still consider the USDC "staking rewards"

Reply:
Of the two options you listed, if it's not too invasive would prefer to rename to `WithdrawalHostDenomBalanceCallback` - these functions were pretty confusing to me so I can only imagine how lost someone who doesn't understand the codebase could be

Reply:
can I address this in a separate PR since it will require a file renaming? I can add it to the canvas

Reply:
Sg! Resolving